### DISCUSSION TITLE
Level Shifter build help

### DISCUSSION THREAD

**Discussion by Coltonc188** [awawa-dev/HyperHDR#995] [Q&A] [answered]
Hello!

I apologize for my ignorance, while I'm technologically inclined, i've never done anything with circuit boards or anything like that.

Sooo, my hue gradient gave out, and am building a Hyperion now, and need a level shifter.

From the documentation I am needing a little bit of help on building it.   
![146442829-b3b81958-e2af-4f85-8002-7458bc9563e5](https://github.com/user-attachments/assets/cb81c626-b7c3-495d-8ae3-e4a228eb10ed)
![146442851-47ff5060-8da1-4f9f-9ce4-cec852a6d775](https://github.com/user-attachments/assets/feb5084c-a362-4b2d-9d1b-b4bd33c4bff6)

These are the two images I am looking at right now.  

Up first on things I need to make it work.  

I was going to get this---https://www.amazon.com/gp/product/B0D8VSYQCW/ref=ox_sc_saved_image_2?smid=A1G9SLP4Q3I0FT&psc=1

to have a bunch of boards and headers, and just incase I needed more for other stuff down the road.

https://www.aliexpress.us/item/2251832650057888.html?spm=a2g0o.cart.0.0.1f5538daXgg01w&mp=1&gatewayAdapt=glo2usa
https://www.aliexpress.us/item/3256805282984717.html?spm=a2g0o.cart.0.0.420938daQ7jhJ6&mp=1&pdp_ext_f=%7B%22cart2PdpParams%22%3A%7B%22pdpBusinessMode%22%3A%22retail%22%7D%7D&gatewayAdapt=glo2usa

I've got this to connect the 2nd item to the board--https://www.amazon.com/gp/product/B09H6TJSJ6/ref=ox_sc_saved_image_3?smid=A27Z2KIJCBIOXA&psc=1

These two to go on the board.  This is where my ignorance kicks in a bit, on the first image, what is it on the PCB board and on the MH-ET LIVE board that connects them?  Can someone link that for me?

Once I have that built, my plan is to use 
https://www.amazon.com/dp/B01N49DSC1?ref_=&colid=9BKX0YHXC5NB&coliid=I2YMZKGFYQUD0B&th=1&psc=1
https://www.amazon.com/dp/B09V1KJ3J4?ref_=&colid=9BKX0YHXC5NB&coliid=I3I02K3D3RQ5U9&th=1
https://www.amazon.com/dp/B07V5JTMV9?ref_=&colid=9BKX0YHXC5NB&coliid=I2RAMY6XB66DVP&th=1

Based on what I have read here that should all work fine does that sound correct? 

Thank you for your time and assistance!!

-Chase

**Comment by awawa-dev** [comment]
Hi Chase
First fundamental question: will you be using wifi for your LED driver, e.g. WLED? If not, why not just buy a Pico(rp2040) board with a level shifter, such as the Adafruit ItsyBitsy (or Scorpio for multi-segment) and use it as a Adalight LED driver? When you add up the cost of components, it can come out to be similar to building a DIY level shifter. https://github.com/awawa-dev/HyperSerialPico?tab=readme-ov-file#recommended-boards-with-a-built-in-level-shifter

**Reply by Coltonc188** [reply]
I can use Wifi, I have an access point right next to where this was going to be setup.  I'm sorry, maybe i'm not doing as well as I thought....why do I need wifi at all?  If everything is processed on the Pi, and this is connected to the pi, and then connected to the TV for the signal.

Am I missing something?

**Reply by awawa-dev** [reply]
Some people prefer wifi because the LED controller is far away from the RPi or they don't want to use cables. However, a cable connection is recommended due to greater stability an in this rp2040 is recommended over esp32. You can connect the Adafruit Scorpio to the USB port on the RPi and its data signal pin to the LEDs. The LED strip ground must have also a connection to the Scorpio ground.

**Reply by Coltonc188** [reply]
Ohh I got ya, I basically planned to mount the Pi to the back of my TV.  I can do whichever you recommend, just trying to get this thing done in the best/most recommended way.

**Reply by awawa-dev** [reply]
I think the easiest way is to buy a ready-made Adafruit rp2040 with a level shifter: it only costs $10 or $15 depending on the model and you will save yourself a lot of time. In the case of ESP32, there is also the lottery of which UART you will get on the board: one that supports 2Mb or the limited cp2102 from the warehouse shelves. Pico also generates a better quality signal than esp32, although for LEDs it does not make a big difference.

**Reply by Coltonc188** [reply]
So something like this? https://www.adafruit.com/product/5724

So assuming I go that route now, how does that change my build?

**Reply by awawa-dev** [reply]
No, this is a wrong model. Search there for "Adafruit ItsyBitsy RP2040" or "Adafruit Scorpio RP2040"
You still need a grabber, LED strip, LED power supply, RPi4 (not sure about linked offer, radiators are not necessary, properly configured HyperHDR is using usually less than 10% of CPU) + RPi power supply. If your external sources are using HDCP protected stream (like Netflix) or Dolby Vision then also Ezcoo splitter will be necessary.

**Reply by Coltonc188** [reply]
If I go with Elgato HD60 X will that work for a grabber and splitter for HDCP devices as well?

**Reply by awawa-dev** [reply]
No modern grabber will bypass HDCP protections (licensing requirements). The only option left is the HDMI splitter.

**Reply by Coltonc188** [reply]
oh bummer....so there really is no reason to go with the Elgato HD60 X then....Sounds like I'm better off just going with the MS2109  and getting the Ezcoo EZ-SP12H2 right?

**Reply by awawa-dev** [reply]
MS2109 is only USB2.0. MS2130 is USB3.0 (but a firmware update may be neccesery). Or latest and greatest UGreen grabber with P010 support: https://www.amazon.com/dp/B0D4LV836Z (although I see that it is terribly expensive in the US, in Europe it costs now about 70 euros).

**Reply by Coltonc188** [reply]
ok, so if I did the MS2130 or the UGreen, then the Ezcoo EZ-SP12H2 I should be good.

The RP2040, what do I need to do with that? just connect the USB to the Pi?

**Reply by awawa-dev** [reply]
> ok, so if I did the MS2130 or the UGreen, then the Ezcoo EZ-SP12H2 I should be good.

just remember that MS2130 and Ezcoo EZ-SP12H2 does not support VRR if you need it.  ‎EZ-SP12H21 and ugreen should support it.

> The RP2040, what do I need to do with that? just connect the USB to the Pi?

yes, and the output data line and the ground to the LED strip (for the common ground with the LED power supply).

**Reply by awawa-dev** [reply]
https://github.com/satgit62/How-to-Install-and-set-up-Ambilight-on-LG-webOS?tab=readme-ov-file#rp2040-adalight-feather-scorpio-mit-level-shifter-ausgang-gpio-16

in your case you connect Scorpio to the RPi, not the LG TV.

the connection to the leds does not have to be in the corner, it can be e.g. in the middle of the lower edge. it does not matter, everything can be configured later in HyperHDR

**Reply by Coltonc188** [reply, marked as answer]
Sorry for the delay, I got off work, and kids got home and they pulled me from my computer....anyways.

Looking at that diagram you put up there.

adding this as power supply, https://www.amazon.com/Single-Output-Enclosed-Switching-Supply/dp/B018TDT20I
then this as a plug to plug into power supply--https://www.amazon.com/QTEATAK-Extension-2-Pole-3-Wire-Straight/dp/B0CHY8NX13/ref=sr_1_3_sspa?crid=3J3IYGI9JSHVQ&dib=eyJ2IjoiMSJ9.Ad_NfMlJTtM5qCKw9LlzRqUwrAu9DYyuMbf5jLrhtDpPEiWOFvSXJAbcltgdjqFJcRZZd6iEJSHwBblC6W0nxl49bpiSBBc0SuJIsUxQiOADZeDuyqbGyFVNHyu4TT29DG-2XxGQl-Pwy_siU5QtFSshB_XC1UVP5V2fRSWwbXFf5e1FoogP7UH1yL8s8CMBGBuqNXf5z2aYvnUKarPVeM_e_9xee551qdjNq9Xie66AhYrBsR-kLQHNFH3WS3d24CIT3XY4SrKYfp-4PE2jk2Ga3bFqyAY7tKmbPSJY1WA.c_xJBvi1DV_Miu1SQlTyNe6dOssoDpg7N7bnJIG2620&dib_tag=se&keywords=plug%2Bwire%2Bwith%2Bground&qid=1732635540&sprefix=plug%2Bwire%2Bwith%2Bground%2Caps%2C230&sr=8-3-spons&sp_csd=d2lkZ2V0TmFtZT1zcF9hdGY&th=1

Now recommendation on the connection between the mean well and the strip, looks like some kind of fuses setup there?  Also this is going on a 75 inch TV, so it looks like I will need to connect this in 3 spots on the TV/LED Lights? 
![2024-11-26_9-44-32](https://github.com/user-attachments/assets/d6f8c884-40b5-4ddf-b903-94ba02c21fff)

**Reply by Coltonc188** [reply]
actually would it matter if the power supply was slightly bigger such as MEAN WELL LRS-100-5?

**Reply by awawa-dev** [reply]
A more powerful power supply won't hurt, I use the over-shot Mean Well LRS-150F-5 myself. A power supply like this also needs a case and preferably an IEC socket with a fuse: safety first.

**Reply by Coltonc188** [reply]
Ok I'll look for that. I'll go with the stronger power supply.

Do you hsve any recommendations on a good brand/company to go with on the case. A good size for our application? I looked at the brand you mentioned in the build. Looks like only in the UK.

Did you see that Pic I loaded above? Does that look accurate?

**Reply by awawa-dev** [reply]
> Looks like only in the UK.

The (local) company is from Poland. I don't know the equivalent from the US market. I'm not sure if you need a third injection point in the middle: once you measure how much the whole thing will come out and cut it off, you can measure the voltage in the middle and check if it's within the norms, e.g.: https://www.pololu.com/file/0J1233/sk6812_datasheet.pdf (+3.5～+5.5) You can ask the BTF manufacturer for their recommendations/specification (if they have any...).
